tools/PI/DevHome.PI/ViewModels/InsightsPageViewModel.cs

```diff
@@ -61,17 +53,19 @@ private void TargetApp_PropertyChanged(object? sender, PropertyChangedEventArgs
 
     internal void AddInsight(Insight insight)
     {
+        var barWindow = Application.Current.GetService<PrimaryWindow>().DBarWindow;
+        Debug.Assert(barWindow != null, "BarWindow should not be null.");
```

Comment:
COnsidering this assert is looking at something in `PrimaryWindow` the assert seems a better fit to move into `PrimaryWindow` to make sure `DBarWindow` isn't null.

Reply:
moved the code out to InsightsService and removed DBarWindow reference

Reply:
Much better. Thank you!